### PR TITLE
Switch k8s tests and origin tests to use rhel9

### PR DESCRIPTION
**Pull request by soltysh**
Needed to test https://github.com/openshift/kubernetes/pull/1954 and https://github.com/openshift/origin/pull/28733

/assign @ardaguclu @joepvd

### CODE REVIEW

**Comment by joepvd** [comment]
/lgtm
yet:
/hold
because merge should be coordinated

File: ci-operator/config/openshift/origin/openshift-origin-master.yaml [hunk 1]
```diff
@@ -11,14 +11,10 @@ base_images:
     name: "4.16"
     namespace: ocp
     tag: hypershift-operator
-  ocp_4.14_base:
```
Comment (droslean) [inline ci-operator/config/openshift/origin/openshift-origin-master.yaml:14]: @soltysh This is being used by:  https://github.com/openshift/release/blob/master/ci-operator/config/openshift/origin/openshift-origin-master.yaml#L35-L37

Comment (ardaguclu) [reply ci-operator/config/openshift/origin/openshift-origin-master.yaml:14]: @droslean this https://github.com/openshift/release/pull/51289 will fix

**Pull request merged**